model/renderNativeWindow.cc

```diff
@@ -40,26 +40,33 @@ namespace minsky
 {
   void RenderNativeWindow::renderFrame(unsigned long parentWindowId, int offsetLeft, int offsetTop, int childWidth, int childHeight)
   {
-    // bool isInitial = false;
     if (!(winInfoPtr.get())) {
-      // isInitial = true;
       winInfoPtr = std::make_shared<WindowInformation>(parentWindowId, offsetLeft, offsetTop, childWidth, childHeight);
     }
+    if(winInfoPtr->getRenderingFlag()) {
+      return;
+    }
+
+    winInfoPtr->setRenderingFlag(true);
```

Comment:
Given this server is single threaded, is this flag logic necessary?

Reply:
Indeed, the flag is not necessary for single threaded application.